TypeScript/disable.ts

```diff
@@ -16,23 +17,46 @@
  * })
  * ```
  */
-import { CommandType, EventPlugin, PluginType } from "@sern/handler";
-import { InteractionReplyOptions, ReplyMessageOptions } from "discord.js";
+import { CommandType, CommandControlPlugin, controller } from "@sern/handler";
+import {
+	InteractionReplyOptions,
+	Message,
+	ReplyMessageOptions,
+} from "discord.js";
 
 export function disable(
 	onFail?:
 		| string
 		| Omit<InteractionReplyOptions, "fetchReply">
 		| ReplyMessageOptions
-): EventPlugin<CommandType.Both> {
-	return {
-		type: PluginType.Event,
-		description: "Disables command from responding",
-		async execute([ctx], controller) {
-			if (onFail !== undefined) {
-				await ctx.reply(onFail);
+) {
+	return CommandControlPlugin<CommandType.Both>(async (ctx, [args]) => {
+		if (onFail !== undefined) {
+			switch (args) {
+				case "text":
+					ctx.reply(onFail)
+						.then((m) => {
```

Comment:
the formatting here is a bit odd, could u compact and fix it?
https://matcha.fyi/pyramid-of-doom-javascript/